gitbootcamp.rst

```diff
@@ -0,0 +1,268 @@
+.. _gitbootcamp:
+
+Git Bootcamp and Cheat Sheet
+============================
+
+In this section, we'll go over some commonly used git commands that are
+relevant to CPython's workflow.
+
+.. contents::
+
+
+Forking CPython GitHub Repository
+---------------------------------
+
+You'll only need to do this once.
+
+1. Go to https://github.com/python/cpython.
+
+2. Press ``Fork``.
+
+3. When asked where to fork the repository, choose to fork it to your username.
+
+4. A fork will be created at https://github.com/UserName/cpython.
+
+
+Cloning The Forked CPython Repository
+-------------------------------------
+
+From your command line::
+
+   $ git clone git@github.com:UserName/cpython.git
+   $ cd cpython
+   $ git remote add upstream git@github.com:python/cpython.git
+
+
+Listing the Remote Repositories
+-------------------------------
+
+::
+
+   $ git remote -v
+
+
+Creating and Switching Branches
+-------------------------------
+
+.. note::
+   Never commit directly to the `master` branch.
+
+Create a new branch::
+
+   $ git checkout -b some-branch master # creates a new branch off master
+
+which is equal to::
```

Comment:
Might want to mention it's equivalent *if* you are starting from the master branch, otherwise explicitly add the `master` part to the `git branch` step.